contracts/epochs/FlowDKG.cdc

```diff
@@ -100,6 +90,230 @@ access(all) contract FlowDKG {
+    access(all) view fun isValidGroupKey(_ groupKey: String?): Bool {
+         if groupKey == nil {
+            // This is a nil/empty submission (see isValidNilSubmission)
+            return true
+        }
+        return groupKey!.length == FlowDKG.submissionKeyLength
+    }
+
+    // Checks a list of participant public keys (part of a ResultSubmission) for validity.
+    // A valid public key in this context is either: (1) a hex-encoded 96 bytes, or (2) nil.
+    access(all) view fun isValidPubKeys(_ pubKeys: [String]?): Bool {
+        if pubKeys == nil {
+            // This is a nil/empty submission (see isValidNilSubmission)
+            return true
+        }
+        for key in pubKeys! {
+            // keys must be exactly 96 bytes
+             if key.length != FlowDKG.submissionKeyLength {
+                 return false
+             }
+        }
+        return true
+    }
+
+    // Checks that an id mapping (part of ResultSubmission) contains one entry per public key.
+    access(all) view fun isValidIDMapping(pubKeys: [String]?, idMapping: {String: Int}?): Bool {
+        if pubKeys == nil {
+            // This is a nil/empty submission (see isValidNilSubmission)
+            return true
+        }
+        return pubKeys!.length == idMapping!.keys.length
+    }
+
+    // ResultSubmission represents a result submission from one DKG participant.
+    // Each submission includes a group public key and an ordered list of participant public keys.
+    // A submission may be empty, in which case all fields are nil - this is used to represent a local DKG failure.
+    // All non-empty submission will have all non-nil, valid fields.
+    // By convention, all keys are encoded as lowercase hex strings, though it is not enforced here.
+    //
+    // INVARIANTS:
+    //  (1) either all fields are nil (empty submission) or no fields are nil
+    //  (2) all key strings are the expected length for BLS public keys
+    //  (3) all non-empty submissions have one participant key per node ID in idMapping
+    access(all) struct ResultSubmission {
+        // The group public key for the beacon committee resulting from the DKG.
+        access(all) let groupPubKey: String?
+        // An ordered list of individual public keys for the beacon committee resulting from the DKG.
+        access(all) let pubKeys: [String]?
+        // A mapping from node ID to DKG index.
+        // There must be exactly one key per authorized DKG participant.
+        // The set of values should form the set {0, 1, 2, ... n-1}, where n is the number of
+        // authorized DKG participant, however this is not enforced here.
+        access(all) let idMapping: {String:Int}?
+
+        init(groupPubKey: String?, pubKeys: [String]?, idMapping: {String:Int}?) {
+            pre {
+                FlowDKG.isValidNilSubmission(groupPubKey: groupPubKey, pubKeys: pubKeys, idMapping: idMapping): "invalid empty submission"
```

Comment:
```suggestion
                FlowDKG.isValidNilSubmission(groupPubKey: groupPubKey, pubKeys: pubKeys, idMapping: idMapping): 
                "FlowDKG.ResultSubmission.init: Invalid DKG Result Submission! "
                .concat("DKG fields must all be `nil` or all be non-`nil`")
```
I'm going through all the core contracts repos and updating the error messages to be more helpful, so it would be good to start doing this for all our PRs from now on. Basically going off the guidelines in [this issue](https://github.com/onflow/docs/issues/795). Can you do something similar for the other panics in this PR?

Reply:
I updated all the error messages in the contract, in https://github.com/onflow/flow-core-contracts/pull/441/commits/ed047b9ea0c205bfa841857148169f595dbf163c